datajunction-server/datajunction_server/internal/authentication/basic.py

```diff
@@ -71,18 +71,21 @@ async def parse_basic_auth_cookie(
         or False
     )
     _logger.info("Attempting to get basic authenticated user from request cookie")
-    jwt = None
+    auth_token = None
     try:
-        jwt = get_jwt(request=request)
-    except (JWTError, AttributeError):
-        pass
-    encrypted_username = jwt.get("sub") if jwt else None
-    username = decrypt(encrypted_username) if encrypted_username else None
+        if any(request.url.path.startswith(endpoint) for endpoint in SSE_ENDPOINTS):
+            auth_token = request.query_params.get("token")
+        else:
+            auth_token = await get_token(request.headers.get("Authorization"))
+    except (JWTError, AttributeError) as exc:
+        _logger.error(str(exc))
+    username = decrypt(auth_token) if auth_token else None
+    _logger.info("User detected as %s", username)
     user = None
     try:
         user = session.exec(select(User).where(User.username == username)).one()
     except NoResultFound:
-        pass
+        _logger.error("Cannot find user %s", username)
```

Comment:
I think you need to do more cleaning here, otherwise `user` might be undefined in line 90 and will raise another exception. We probably need to set `user = None` here too.

Alternatively you can do `one_or_none()` in line 86, and remove the `try/except` block, I think.

Reply:
I've broken this out to a `get_user` function that uses `one_or_none()` and raises a DJException when a user can't be found. This feels cleaner since we can now easily differentiate between the user not existing and the password being incorrect. (My understanding though is there's a benefit to obscuring that to clients, something about a timing attack)